### PR TITLE
Jetpack Sync: First pass at functional UI

### PR DESCRIPTION
**Pull request by ebinnion**
This PR is a first pass at making the sync status UI functional. At this point, you should be able to go to `/settings/general/$site` and click the "Perform Full Sync" button in order to start a sync.

After starting a sync, you should see a status bar that updates, and when the sync is finished, the UI should reflect that.

cc @lezama for code review and @rickybanister for any early design feedback.

To test:

- Checkout `update/jetpack-sync-status-hook-up` branch
- Make sure Jetpack is on latest of `branch-4.2`
- Go to `/settings/general/$site`
- Click "Perform full sync" button
- Verify full sync is performed by going to `$site/wp-admin/admin.php?page=jetpack-sync`
- Verify progress bar is removed from UI when sync is over

Known issues:

- This PR does not handle the UI for errors during full sync. For example, if you lose internet and then internet comes back.
- <strike>This PR does not yet handle the informational notice. It is still simply a placeholder.</strike>

Test live: https://calypso.live/?branch=update/jetpack-sync-status-hook-up

### CODE REVIEW

**Comment by rickybanister** [comment]
This looks good so far @ebinnion.

A couple of things I noticed (and already mentioned to you):

- [ ] We should link to a support document that further explains Sync. I'll help figure out if that should be more than just a text link in the description.
- [ ] Notices with no status used in this context should have no background and be `$gray` text
- [ ] We can remove the old staging-only sync setting box
- [ ] If a sync is in progress it might be helpful to other users to show who initiated it and the start time. Something like `Eric started a full sync 27 minutes ago` in the notice. If you are the one that initiated the sync, we can keep it as planned `Performing full sync` 

I think that's all I can think of for now.

File: client/my-sites/site-settings/jetpack-sync-panel/index.jsx [hunk 1]
```diff
@@ -24,27 +30,67 @@ const debug = debugModule( 'calypso:site-settings:jetpack-sync-panel' );
...
 
 	renderStatusNotice() {
-		const classes = classNames( 'jetpack-sync-panel__notice' );
+		const finished = get( this.props, 'syncStatus.finished' );
+		const finishedTimestamp = this.moment( parseInt( finished, 10 ) * 1000 );
+		let text = '';
+		if ( this.shouldDisableSync() ) {
+			text = this.translate( 'Full sync in progress' );
+		} else if ( finishedTimestamp.isValid() ) {
+			text = this.translate( 'Last syned %(ago)s', {
```
Comment (lezama) [inline client/my-sites/site-settings/jetpack-sync-panel/index.jsx]: Last syned -> Last synced

**Comment by lezama** [comment]
LGTM :shipit:

**Comment by ebinnion** [comment]
At this point in the PR:

- I fixed an issue where the progress bar would load and then disappear. The UI should be functional now
- Notice styling has been updated
- Now using `Interval` component for polling as opposed to manually handling logic with Lodash's `throttle`.
- Sync panel now displayed in Jetpack panel

I think we're ready to go ahead and get this PR in for now, and I can address other feedback in follow-up PRs.

@lezama - Do you know if we can go ahead and remove the non-public post types checkbox? Also, do you have a preferred for sync?

@rickybanister - We don't currently store the data on who started the last full sync. We can maybe get that information in this release, but now that Jetpack is in beta, I'm inclined to push it.

**Comment by rickybanister** [comment]
@ebinnion correct me if I'm wrong, but I thought I saw your progress bar when I viewed my settings. My suggestion about displaying the user that initiated the sync was to alleviate confusion while one is running. I think it's less important to display who ran the last one after the fact. 

Can't we just store on the Calypso side who clicked the button rather than doing that in Jetpack?

**Comment by lezama** [comment]
> I think we're ready to go ahead and get this PR in for now, and I can address other feedback in follow-up PRs.

agreed.

>  Do you know if we can go ahead and remove the non-public post types checkbox?

Let's do it for sites running > 4.1.1

**Comment by ebinnion** [comment]
@rickybanister - It's a bit more than that. Storing on the Calypso side would only store it in the current user's browser. So, we would need to store "who" initiated the sync on the Jetpack side so that we could persist and distribute that information.

I can create separate issues for your feedback and then make passes at them in separate PRs.

**Pull request merged**